### PR TITLE
ea-utils: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package ea-utils

### CODE REVIEW

File: var/spack/repos/builtin/packages/ea-utils/package.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    depends_on('bamtools')
+    # perl module required for make check
+    depends_on('perl', type='build')
+
+    build_directory = 'clipper'
+
+    def edit(self, spec, prefix):
+        with working_dir('clipper'):
+            makefile = FileFilter('Makefile')
+            makefile.filter('/usr', self.spec.prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ea-utils/package.py]: `self.spec.prefix == prefix`. It's probably better to use the latter.

File: var/spack/repos/builtin/packages/ea-utils/package.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    url      = "https://github.com/ExpressionAnalysis/ea-utils/archive/1.04.807.tar.gz"
+
+    version('1.04.807', '5972b9f712920603b7527f46c0063a09')
+
+    depends_on('subversion')
+    depends_on('zlib')
+    depends_on('gsl')
+    depends_on('bamtools')
+    # perl module required for make check
+    depends_on('perl', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ea-utils/package.py]: We don't usually add dependencies that are only needed for testing. See #1279. Until that is solved, I've been adding:
```python
# TODO: Add a 'test' deptype                                                
# https://github.com/LLNL/spack/issues/1279                                 
# depends_on('perl', type='test')
```
to these kinds of packages.

Comment (akthoma) [inline var/spack/repos/builtin/packages/ea-utils/package.py]: For now, though, am I able to add those comments to the package but still keep the `depends_on('perl', type='build')` that I have now? The package will not build without it.

Comment (akthoma) [inline var/spack/repos/builtin/packages/ea-utils/package.py]: Correct. The build fails with this error when running `make`:
```
prove -j 4 t
Can't locate Test/Builder.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/join.t line 1.
BEGIN failed--compilation aborted at t/join.t line 1.
t/join.t ... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Can't locate Test/Builder.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/mcf.t line 1.
BEGIN failed--compilation aborted at t/mcf.t line 1.
t/mcf.t .... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Can't locate Test/Builder.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/multx.t line 1.
BEGIN failed--compilation aborted at t/multx.t line 1.
t/multx.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 

Test Summary Report
-------------------
t/join.t (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
t/mcf.t  (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
t/multx.t (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
Files=3, Tests=0,  1 wallclock secs ( 0.02 usr +  0.00 sys =  0.02 CPU)
Result: FAIL
make: *** [check] Error 1
```

In the `Makefile`, we have:
`all: $(BIN) check` at the beginning, and later in the file:
`check: $(BIN)
        prove -j 4 t` of which will not work without perl

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ea-utils/package.py]: I see. That's a poorly designed Makefile if you aren't able to build without testing. Those steps are always separate. Maybe you should add a comment like:
```python
# perl module required for make check, which is included in the default target
```

File: var/spack/repos/builtin/packages/ea-utils/package.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+from spack import *
+
+
+class EaUtils(MakefilePackage):
+    """Command-line tools for processing biological sequencing data. Barcode
+       demultiplexing, adapter trimming, etc. Primarily written to support an
+       Illumina based pipeline - but should work with any FASTQs."""
+
+    homepage = "http://expressionanalysis.github.io/ea-utils/"
+    url = "https://github.com/ExpressionAnalysis/ea-utils/archive/1.04.807.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ea-utils/package.py]: For the record, we ignore lines that look like URLs when doing line-length checks. You can see a list of what we ignore [here](https://github.com/LLNL/spack/blob/develop/.flake8) and [here](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L54).

**Pull request merged**